src/swap/SwapScreen.tsx

```diff
@@ -100,22 +115,28 @@ export function SwapScreen() {
       setFromSwapAmountError(true)
       dispatch(showError(t('swapScreen.insufficientFunds', { token: fromToken?.symbol })))
     } else {
-      navigate(Screens.SwapReviewScreen, {
-        toToken: toToken!.address,
-        fromToken: fromToken!.address,
-        swapAmount,
-        updatedField,
-      })
+      dispatch(
+        setSwapUserInput({
+          toToken: toToken!.address,
+          fromToken: fromToken!.address,
+          swapAmount,
+          updatedField,
+        })
+      )
+      navigate(Screens.SwapReviewScreen)
     }
   }
 
   const handleShowTokenSelect = (fieldType: Field) => () => {
     ValoraAnalytics.track(SwapEvents.swap_screen_select_token, { fieldType })
-    Keyboard.dismiss()
     // ensure that the keyboard is dismissed before animating token bottom sheet
-    setTimeout(() => {
-      setSelectingToken(fieldType)
-    }, 100)
+    Keyboard.addListener('keyboardDidHide', onKeyboardDidHide(fieldType))
```

Comment:
i'm confused, we are removing the keyboard listener inside the `addListener`? why do we need this?

Reply:
This is to ensure the setSelectingToken is called after the keyboard has been fully removed. There's probably a better way but using setTimeout with a value of 100 can result in the swap review button appearing in the wrong place on the screen on Android devices.

Reply:
On Android with the setTimeout code, complete a swap and then tap 'Swap Again' Then observe that the 'Review' button will not be aligned with the bottom of the page.

Reply:
There is probably a better way to do this; I'll see what I can come up with!

Reply:
Wasn't able to find a better way but as we are adding and quickly removing the listener this seems fine after talking with @jeanregisser.